plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx

```diff
@@ -13,371 +13,97 @@
+  const updateLocalState = useCallback(
+    (newState: RepoUrlPickerState) => {
+      setState(prevState => ({ ...prevState, ...newState }));
+    },
+    [setState],
+  );
 
   return (
     <>
-      <FormControl
-        margin="normal"
-        required
-        error={rawErrors?.length > 0 && !host}
-      >
-        <Select
-          native
-          disabled={hostsOptions.length === 1}
-          label="Host"
-          onChange={updateHost}
-          selected={host}
-          items={hostsOptions}
+      <RepoUrlPickerHost
+        host={state.host}
+        hosts={allowedHosts}
+        onChange={host => setState(prevState => ({ ...prevState, host }))}
+        rawErrors={rawErrors}
+      />
+      {state.host && integrationApi.byHost(state.host)?.type === 'github' && (
+        <GithubRepoPicker
+          allowedOwners={allowedOwners}
+          rawErrors={rawErrors}
+          state={state}
+          onChange={updateLocalState}
         />
-
-        <FormHelperText>
-          The host where the repository will be created
-        </FormHelperText>
-      </FormControl>
-      {/* Show this for dev.azure.com only */}
-      {host === 'dev.azure.com' && (
-        <FormControl
-          margin="normal"
-          required
-          error={rawErrors?.length > 0 && !organization}
-        >
-          <InputLabel htmlFor="repoInput">Organization</InputLabel>
-          <Input
-            id="repoInput"
-            onChange={updateOrganization}
-            value={organization}
-          />
-          <FormHelperText>The name of the organization</FormHelperText>
-        </FormControl>
       )}
-      {host && integrationApi.byHost(host)?.type === 'bitbucket' && (
-        <>
-          {/* Show this for bitbucket.org only */}
-          {host === 'bitbucket.org' && (
-            <FormControl
-              margin="normal"
-              required
-              error={rawErrors?.length > 0 && !workspace}
-            >
-              <InputLabel htmlFor="wokrspaceInput">Workspace</InputLabel>
-              <Input
-                id="wokrspaceInput"
-                onChange={updateWorkspace}
-                value={workspace}
-              />
-              <FormHelperText>
-                The workspace where the repository will be created
-              </FormHelperText>
-            </FormControl>
-          )}
-          <FormControl
-            margin="normal"
-            required
-            error={rawErrors?.length > 0 && !project}
-          >
-            <InputLabel htmlFor="wokrspaceInput">Project</InputLabel>
-            <Input
-              id="wokrspaceInput"
-              onChange={updateProject}
-              value={project}
-            />
-            <FormHelperText>
-              The project where the repository will be created
-            </FormHelperText>
-          </FormControl>
-        </>
+      {state.host && integrationApi.byHost(state.host)?.type === 'gitlab' && (
```

Comment:
feels like we need a `const hostType = state.host && integrationApi.byHost(state.host)?.type || undefined` or something at the top